pkg/operator/resource/resourceapply/core.go

```diff
@@ -3,42 +3,102 @@ package resourceapply
 import (
 	"bytes"
 	"context"
+	"crypto/md5"
 	"fmt"
+	"io"
 	"sort"
 	"strings"
 
-	"k8s.io/klog/v2"
-
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/equality"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	coreclientv1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/klog/v2"
 
 	"github.com/openshift/library-go/pkg/operator/events"
 	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
 )
 
+type CachedVersionKey struct {
+	name, kind, namespace string
+}
+
+type CachedResource struct {
+	resourceHash, resourceVersion string
+}
+
+// map of name,kind to resourceHash (of last applied resource) : resourceVersion(when resource was applied)
+var cachedResourceVersions map[CachedVersionKey]CachedResource = make(map[CachedVersionKey]CachedResource)
```

Comment:
no global state please.  I'd rather either
1. break everyone and change the methods
2. create new methods that accept a cache or use a method receiver struct to track it and have the staticresource controller maintain them.

Reply:
I chose option 2. Since there are so many users of resourceapply.ApplyFoo functions, I decided to create new functions that accept a cache and use those new functions from static resource controller, buta t some point I'd like to go back and add something that lets everyone take advantage of the enhanced performance.